### ISSUE TITLE
ext_event example panics on linux

### ISSUE THREAD

**Issue by kindlychung** [xi-editor/druid#594]
Here is the error message:

```
DEBUG [druid::localization] available locales [fr-CA, de-DE, en-US], current en-US
DEBUG [druid::localization] resolved: [en-US]
INFO  [druid_shell::platform::gtk::application] gtk: Activated application
thread '<unnamed>' panicked at 'GDK may only be used from the main thread.', <::std::macros::panic macros>:2:4
```

**Comment by ForLoveOfCats** [comment]
Can confirm on Manjaro GNU/Linux with GTK 3.24.14 on master (currently 000d554cdd06887c750658794dfe97495ff70362)
Backtrace: https://gist.github.com/ForLoveOfCats/7146f826205937ed7a2738d4c84a992b

**Comment by raphlinus** [comment]
I think this might be a bug in gtk-rs. Looking at the [source](http://gtk-rs.org/docs/src/gdk/auto/functions.rs.html#545-558), it seems like we're *correctly* calling the function that maps to `gdk_threads_add_idle` (see https://developer.gnome.org/gdk3/stable/gdk3-Threads.html) but there seems to be an overly aggressive assert that we're on the main thread, even though I believe the `threads` functions don't require that.

File an issue on gtk-rs, and they can either confirm my analysis, or tell us what we're doing wrong? This seems like something that other people are likely to have bumped into.

**Comment by cmyr** [comment]
this is fixed in #550, just waiting on review.
